### ISSUE TITLE
Cannot install in Ubuntu 24.04 LTS, GNOME 46

### ISSUE THREAD

**Issue by sfa77** [yuezk/GlobalProtect-openconnect#351]
**Describe the bug**
Cannot install the package in Ubuntu 24.04 LTS, GNOME 46

**Logs**
sudo apt-get install globalprotect-openconnect
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
      globalprotect-openconnect : Depends: libjavascriptcoregtk-4.0-18 but it is not 
                                                          Depends: libwebkit2gtk-4.0-37 (>= 2.21.1) but it is not installable
E: Unable to correct problems, you have held broken packages.

**Environment:**
 - OS: Ubuntu 24.04 LTS
 - Desktop Environment: GNOME

**Comment by eitzenbe** [comment]
I am affected by this too, seems the lib versions have moved up on ubuntu 24.04

**Comment by yuezk** [comment]
Hi @sfa77 @eitzenbe [`libjavascriptcoregtk-4.0-18`](https://answers.launchpad.net/ubuntu/noble/amd64/libjavascriptcoregtk-4.0-18/2.43.3-1) and [`libwebkit2gtk-4.0-37`](https://answers.launchpad.net/ubuntu/noble/amd64/libwebkit2gtk-4.0-37/2.43.3-1) were deleted (status is `Deleted`) in the Ubuntu 24.04 repository. Current workaround is to download and install these packages manually before installing `globalprotect-openconnect`.

```sh
wget http://launchpadlibrarian.net/704701349/libwebkit2gtk-4.0-37_2.43.3-1_amd64.deb
wget http://launchpadlibrarian.net/704701345/libjavascriptcoregtk-4.0-18_2.43.3-1_amd64.deb

sudo dpkg --install *.deb
```

**Comment by eitzenbe** [comment]
THANX - installation succeeds and gui comes up but issues an error:

[2024-04-28T08:09:39Z WARN  gpgui::app::app_initializer] Can't read the config key from the keyring: Couldn't access platform secure storage: SS error: result not returned from SS API
[2024-04-28T08:09:40Z WARN  gpapi::utils::window] Failed to raise window: Failed to raise window: GlobalProtect
[2024-04-28T08:09:46Z INFO  gpgui::config::private_data] Generating config key
[2024-04-28T08:09:46Z WARN  gpgui::app::app_initializer] Can't read the config key from the keyring: Couldn't access platform secure storage: SS error: result not returned from SS API

step by step :)

**Comment by yuezk** [comment]
@eitzenbe, I encountered this problem yesterday and it is back to normal after a restart. And what' the output of `ps aux | grep 'gnome-keyring\|kwalletd5' | grep -v grep`?

**Comment by eitzenbe** [comment]
> ps aux | grep 'gnome-keyring\|kwalletd5' | grep -v grep
.........  /usr/bin/gnome-keyring-daemon --foreground --components=pkcs11,secrets --control-directory=/run/user/1000/keyring

**Comment by eitzenbe** [comment]
yields /usr/bin/gnome-keyring-daemon --foreground --components=pkcs11,secrets --control-directory=/run/user/1000/keyring
as local user, fwiw: I hadn't set up my wallet in KDE, did that, rebooted, but still same error

**Comment by eitzenbe** [comment]
https://github.com/microsoft/vscode/issues/187380

seems to be related?

**Comment by yuezk** [comment]
Yes. should be the same problem, but this client currently doesn't support the file keychain. Others encountered this issue and resolved it in this comment in https://github.com/yuezk/GlobalProtect-openconnect/issues/321#issuecomment-1938876002 Not sure whether it works for you.

And what's your desktop environment, KDE?

Anyway, I will add the file-based fallback keychain soon.

**Comment by eitzenbe** [comment]
not an expert on that gnome-keyring/kwallet issue, but tried and failed, running KDE ... Any timeline for the "support on filebased" ? No pressure here, THANX for the quick support so far and the work you did!

**Comment by yuezk** [comment]
It should be quick, I'm planning to release a new version with several enhancements.

**Comment by yuezk** [comment]
Add the link to the official issue: https://bugs.launchpad.net/ubuntu/+source/webkit2gtk/+bug/2061914

**Comment by yuezk** [comment]
I'm afraid it is impractical to ask the Ubuntu team to reintroduce `webkit2gtk-4.0`.

Tauri 1.x, which depends on `webkit2gtk-4.0`. Tauri started to use `webkit2gtk-4.1` in 2.0, but it has not been released yet. Since few breaking changes exist between `webkit2gtk-4.1` and `webkit2gtk-4.0`, I plan to port the Tauri 1.x to use `webkit2gtk-4.1`. I have managed to make this work but with one error. And I'm currently working on it.

**Comment by yuezk** [comment]
> not an expert on that gnome-keyring/kwallet issue, but tried and failed, running KDE ... Any timeline for the "support on filebased" ? No pressure here, THANX for the quick support so far and the work you did!

Hi @eitzenbe it has been fixed in 2.2.0, you can follow this instruction to upgrade it in 24.04.

https://github.com/yuezk/GlobalProtect-openconnect?tab=readme-ov-file#ubuntu-2404

**Comment by joca-bt** [comment]
This can also be resolved by upgrading the dependencies of GlobalProtect-openconnect to 4.1. Is there any plan for that?

**Comment by joca-bt** [comment]
The solution on https://github.com/yuezk/GlobalProtect-openconnect?tab=readme-ov-file#ubuntu-2404-and-later no longer works, as there are issues installing these packages due to incompatible dependencies (e.g. libicu).

Couldn't this be resolved by upgrading the versions of these libs that GlobalProtect-openconnect uses? Is there any plan for that?

**Comment by yuezk** [comment]
@joca-bt I plan to upgrade the application when `Tauri 2.0` is released. Which is expected to be released at the end of this month. I'm going to start working on upgrading this application to use Tauri 2.0 soon.

**Comment by sebma** [comment]
@joca-bt What is `Tauri` ?

**Comment by yuezk** [comment]
@sebma It's the [GUI framework](https://tauri.app/) used by this client.

**Comment by sebma** [comment]
@yuezk So once Tauri v2.0 is out, will you release a new `globalprotect-openconnect` package (v2.40 ?) based on `libwebkit2gtk-4.1` and `libjavascriptcoregtk-4.1` in [your launchpad repository](https://launchpad.net/~yuezk/+archive/ubuntu/globalprotect-openconnect) for Ubuntu 24.04 and later ?

**Comment by yuezk** [comment]
@sebma Yes.

**Comment by yuezk** [comment]
> I think that the solution on [yuezk/GlobalProtect-openconnect#ubuntu-2404-and-later](https://github.com/yuezk/GlobalProtect-openconnect?tab=readme-ov-file#ubuntu-2404-and-later) may no longer work, as there are issues installing these packages due to incompatible dependencies (e.g. libicu).

@joca-bt  I tested it and it still works.

**Comment by zaitseff-unsw** [comment]
You might have already seen this, but Tauri 2.0 has just been released...

**Comment by zaitseff-unsw** [comment]
I notice that you've started a `tauri2` branch back on 18th November.  Looking forward to that work being completed, especially for `gpauth`.  Any likely timeline?  I'm not an expert on Rust, but is any help needed?

**Comment by yuezk** [comment]
@zaitseff-unsw Yes, I'm working on it. The expected time could be Jan 10th.

**Comment by yuezk** [comment]
Released in [2.4.0](https://github.com/yuezk/GlobalProtect-openconnect/releases/tag/v2.4.0). I'm closing it. Let me know if you have any questions.
